src/org/labkey/test/tests/list/ListTest.java

```diff
@@ -690,11 +690,67 @@ public void testBatchInsertErrors() throws IOException, CommandException
         // no need to query the list; nothing will be inserted if the batch insert fails/errors
     }
 
+    @Test
+    public void testListMerge()
+    {
+        String mergeListName = "listForMerging";
+        createList(mergeListName, BatchListColumns, BatchListData);
+        _listHelper.beginAtList(PROJECT_VERIFY, mergeListName);
+
+        _listHelper.clickImportData();
+        checker().verifyTrue("For list with an integer, non-auto-increment key, merge option should be available", _listHelper.isMergeOptionPresent());
+        _listHelper.chooseFileUpload();
+        checker().verifyTrue("For list with an integer, non-auto-increment key, merge option should be available", _listHelper.isMergeOptionPresent());
+        _listHelper.chooseCopyPasteText();
+
+        log("Try to upload the same data without choosing to merge.  Errors are expected.");
+        String[] expectedErrors = new String[]{
+                "duplicate key value violates unique constraint"
+        };
+        setListImportAsTestDataField(toTSV(BatchListColumns, BatchListData), expectedErrors);
```

Comment:
Should this error check be done with new data, so a check can be made to ensure no changes were applied?

Reply:
Sure.  Done.